### PR TITLE
KeeShare: Remove checking signed container and QuaZip

### PR DESCRIPTION
**Pull request by droidmonkey**
* Remove QuaZip dependency in favor of minizip
* Remove signature checks, but maintain signatures for backwards compatibility
* Remove UI components related to certificates except for personal certificate for backwards compatibility
* Default to unsigned containers (*.kdbx)

## Screenshots
[TIP]:  # ( Do not include screenshots of your actual database! )
This is the original KeeShare settings in 2.6.6 (showing trusted keys generated in 2.7.0 at the bottom):
image

This is the new settings view in 2.7.0:
image

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Tested KeeShare between 2.7.0 and 2.6.6, worked perfectly. Signatures are still recognized and validated in 2.6.6 so backwards compatibility is maintained. In 2.7.0 we no longer care about verifying the signature.

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Refactor (significant modification to existing code)

### CODE REVIEW

**Comment by droidmonkey** [comment]
Looks like minizip needs some more love on macOS.

File: src/keeshare/ShareExport.cpp [hunk 1]
```diff
@@ -108,124 +106,106 @@ namespace
...
+        zipOpenNewFileInZip64(zf,
+                              fileName.toLatin1().data(),
+                              nullptr,
+                              nullptr,
+                              0,
+                              nullptr,
+                              0,
+                              nullptr,
+                              Z_DEFLATED,
+                              Z_DEFAULT_COMPRESSION,
```
Comment (phoerious) [inline src/keeshare/ShareExport.cpp]: This should be `Z_BEST_COMPRESSION`. There is absolutely no reason to use any compression level other than 9 with GZip/Deflate.

**Review by phoerious — APPROVED** [review]
No major objections. Great net code reduction.

File: src/keeshare/ShareExport.cpp (hunk 1 above)
Comment (droidmonkey) [reply src/keeshare/ShareExport.cpp]: Technically it doesn't matter because you are compressing random data.

**Comment by droidmonkey** [comment]
Need to fix the TestSharing build to include the core library.

**Comment by droidmonkey** [comment]
I ended up reverting the BinaryStream file move and used a QDataStream to write the ssh-rsa key signature file.

**Pull request merged**